src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/HostingAbstractionsHostExtensions.cs

```diff
@@ -104,7 +104,7 @@ public static async Task WaitForShutdownAsync(this IHost host, CancellationToken
             await waitForStop.Task;
 
             // Host will use its default ShutdownTimeout if none is specified.
-            await host.StopAsync();
+            await host.StopAsync(token);
```

Comment:
```suggestion
            // The cancellation token may have been triggered to unblock waitForStop. Don't pass it here because that would trigger an abortive shutdown.
            await host.StopAsync();
```

Reply:
@Tratcher the analyzer will keep getting triggered unless we supress it. There are 3 options:
- use `#pragma warning`
- pass `default`
- pass `CancellationToken.None`
Which do you prefer?

Reply:
`default` looks the cleanest.

Reply:
`default` is the shortest, but it's also the least clear when it comes to *reading* the code. e.g. "Is that `default` supplying no cancellationtoken, or a 0, or a null?" 
So IMO when it's anything less than clear (e.g. the last argument passed to an async method), using `CancellationToken.None` is preferred.

Reply:
I think I am also inclined to use `CancellationToken.None`. If there any objections, @Tratcher @halter73, let me know.

Reply:
I agree that maybe `CancellationToken.None` might be a better choice for this particular instance, also because of [this](https://github.com/dotnet/runtime/pull/37607/files#r444453971).